app/models/tiny_mce_asset.rb

```diff
@@ -65,21 +100,41 @@ def open
     end
   end
 
+  def delete_unsaved_image
+    destroy unless saved
```

Comment:
I think it is better to implement it as class method with asset id as argument and handle inside it situation when this file was already deleted, otherwise it might end up as failed DJ